pylint/checkers/similar.py

```diff
@@ -342,11 +343,17 @@ def __init__(
         ignore_imports: bool = False,
         ignore_signatures: bool = False,
     ) -> None:
-        self.min_lines = min_lines
-        self.ignore_comments = ignore_comments
-        self.ignore_docstrings = ignore_docstrings
-        self.ignore_imports = ignore_imports
-        self.ignore_signatures = ignore_signatures
+        # If we run in pylint mode we link the namespace objects
+        if isinstance(self, BaseChecker):
+            self.namespace = self.linter.config
+        else:
+            self.namespace = argparse.Namespace()
```

Comment:
Shouldn't it be ``self.config`` ?

Reply:
This caused a number of issues with `config` being a property on `BaseChecker` in order to deprecate it. I could create a `setter` but then we would still hit the `DeprecationWarning` every time we accessed it because it is still considered `BaseChecker.config`.

Chose to do it like this, also makes it clear why this is different from `BaseChecker`.

Reply:
All right, let's avoid the headache 😄